flatliners/clusteralertcorr.py

```diff
@@ -0,0 +1,132 @@
+import math
+import pandas as pd
+from datetime import datetime
+from .baseflatliner import BaseFlatliner
+
+class ClusterAlertCorrelation(BaseFlatliner):
+
+    '''
+    This class contains the code for correlation of alerts for every cluster
+
+    '''
+
+    def __init__(self):
+        print(" Inside alert corr")
+        super().__init__()
+        ## Hold the values for different clusters
+        self.clusters = dict()
+        #TODO : save correlation over time
+        self.clusters_correlation = dict()
+        ### To keep track of previous publish
+        self.timestamp = datetime.fromtimestamp(0)
+
+    def on_next(self, x):
+        """ On each data loading it will create dataframe and then pivot frame
+        """
+        # print(f"got {x}")
+
+        if not self.metric_name(x) == 'alerts':
+            return
+
+        alert_name = self.metric_label(x, 'alertname')
+        cluster_id = self.cluster_id(x)
+
+        if cluster_id not in self.clusters:
+            self.clusters[cluster_id] = pd.DataFrame(columns=['_id', 'timestamp', 'alertname'])
+            self.clusters_correlation[cluster_id] = pd.DataFrame()
+
+        previous_df = self.clusters[cluster_id]
+
+        ### Dataframe creation
+        timestamp = []
+        cluster = []
+        alert_name_list = []
```

Comment:
For the consistency comments above - either use `_list` suffix or `s` suffix for vectors (so if you decide to go with `_list` this is fine, otherwise it needs a change too)

Reply:
This will be list

Reply:
I think 's' suffix is maintained for scalars, and in matlab env. and '_v' for vectors.

Reply:
Ahh ok. I thought scalar vs vector. 
Done!